kernel/src/hil/eic.rs

```diff
@@ -0,0 +1,55 @@
+//! Interface for external interrupt controller.
+
+/// Enum for selecting which edge to trigger interrupts on.
+#[derive(Debug)]
+pub enum InterruptMode {
+    RisingEdge,
+    FallingEdge,
+    HighLevel,
+    LowLevel,
+}
+
+/// Enum for enabling/disabling filter
+pub enum FilterMode {
+    FilterEnable,
+    FilterDisable,
+}
+
+/// Enum for selecting syn/asyn mode
+pub enum SynchronizationMode {
+    Synchronous,
+    Asynchronous,
+}
+
+/// Interface for EIC.
+pub trait ExternalInterruptController {
```

Comment:
Would this make sense to be called `DirectInterrupt`? Or perhaps External Interrupt is a more common name than I imagine.